engine/access/index/simple_index.go

```diff
@@ -0,0 +1,106 @@
+package index
+
+import (
+	"fmt"
+
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/module/state_synchronization/indexer"
+	"github.com/onflow/flow-go/storage"
+
+	"github.com/onflow/flow-go/module/state_synchronization"
+)
+
+var _ state_synchronization.IndexReporter = (*SimpleIndex)(nil)
+
+// SimpleIndex implements a wrapper around `IndexReporter` ensuring that needed data has been synced and is available to the client.
+// Note: `SimpleIndex` is created with empty reporter due to the next reasoning:
+// When the index is initially bootstrapped, the indexer needs to load an execution state checkpoint from
+// disk and index all the data. This process can take more than 1 hour on some systems. Consequently, the Initialize
+// pattern is implemented to enable the Access API to start up and serve queries before the index is fully ready. During
+// the initialization phase, all calls to retrieve data from this struct should return indexer.ErrIndexNotInitialized.
+// The caller is responsible for handling this error appropriately for the method.
+type SimpleIndex struct {
```

Comment:
how about `Reporter`. `SimpleIndex` doesn't really communicate what this struct does
```suggestion
type Reporter struct {
```